lib/python/picongpu/plugins/data/base_reader.py

```diff
@@ -32,32 +32,41 @@ def __init__(self, run_directory):
         self.data_file_prefix = None
         self.data_file_suffix = None
 
+        self._update_docstrings()
+
+    def _update_docstrings(self):
+        # only for the derived classes
+        if self.__class__ == DataReader:
+            return
+
+        self.__class__.get.__doc__ = self._get_for_iteration.__doc__
```

Comment:
Should this append or overwrite?
Otherwise LGTM.

Reply:
@ax3l good question - @codingS3b what do you want: overwrite or append?

Reply:
The current implementation fully replaces the docstring, however, the function signature is unchanged. 
![docstring_replace](https://user-images.githubusercontent.com/31093529/74227601-1865a300-4cbf-11ea-8fc8-47a45728808c.png)

The docstring of the base class is this
![docstring_replace_baseclass](https://user-images.githubusercontent.com/31093529/74227885-ac376f00-4cbf-11ea-9fff-1f8b9569d544.png)

So, I would rely on your judgement, whether overwriting or appending would be better.

Reply:
I just discovered that the docstring update has a negative side-effect and probably should not be used and would need modification:
the `self.__class__.get` is a reference to the base classes `get` method and alters the docstring for the base class as well and not only for the derived class for which we want that :(

Reply:
Okay - this is definitely something one should avoid.  WHy do you have to use `self.__class__` anyway? The examples in [the monkey-patch link](https://filippo.io/instance-monkey-patching-in-python/) provided by @ax3l only apply the changes to an instance of the class, not to the class itself?

Reply:
Yes, but as I tried before, the doc attribute is not writeable and can't be patched, that is why I used the class reference.

Reply:
Okay - that's problematic. @codingS3b Do you agree with me that overwriting the documentation of the base class is not an option?

Reply:
Yeah, I would also rather avoid it if possible. However, I don't have a solution available right now :(

Reply:
I will thing about a solution as well. 

cc'ing @BrianMarre and @wmles (they might have some ideas as well)

Reply:
the cleanest thing is to inherit the method and give it a new docstring:
```
def get(*args, **kwargs):
    """ new docstring """
    return super().get(*args, **kwargs)

this is probably slightly more expensive than not inheriting, but since you are using python, I assume this is no issue.

And I assume you can also overwrite the method for the instances inside their __init__ - but does not have any advantage...